spiketoolkit/study/groundtruthstudy.py

```diff
@@ -8,131 +8,139 @@
 
 from spiketoolkit.comparison.groundtruthcomparison import compare_sorter_to_ground_truth, _perf_keys
 
-from .studytools import (setup_comparison_study, run_study_sorters, get_rec_names, 
-                                        get_recordings, copy_sortings_to_npz, iter_computed_names,
-                                        iter_computed_sorting, collect_run_times)
-
-
-
+from .studytools import (setup_comparison_study, run_study_sorters, get_rec_names,
+                         get_recordings, copy_sortings_to_npz, iter_computed_names,
+                         iter_computed_sorting, collect_run_times)
 
 
 class GroundTruthStudy:
     def __init__(self, study_folder=None):
         self.study_folder = Path(study_folder)
-        
+
         self.scan_folder()
         self.computed_names = []
-        
+
         self.comparisons = None
         self.exhaustive_gt = None
-    
+
     def __repr__(self):
         t = 'Groud truth study\n'
-        t += '  ' + str(self.study_folder)+ '\n'
+        t += '  ' + str(self.study_folder) + '\n'
         t += '  recordings: {} {}\n'.format(len(self.rec_names), self.rec_names)
         if len(self.sorter_names):
             t += '  sorters: {} {}\n'.format(len(self.sorter_names), self.sorter_names)
-        
+
         return t
-    
+
     def scan_folder(self):
         self.rec_names = get_rec_names(self.study_folder)
-        
         # scan computed names
-        self.computed_names = list(iter_computed_names(self.study_folder))  # list of pair (rec_name,sorter_name)
+        self.computed_names = list(iter_computed_names(self.study_folder))  # list of pair (rec_name, sorter_name)
         self.sorter_names = np.unique([e for _, e in iter_computed_names(self.study_folder)]).tolist()
 
     @classmethod
     def setup(cls, study_folder, gt_dict):
         setup_comparison_study(study_folder, gt_dict)
         return cls(study_folder)
-    
+
     def run_sorters(self, sorter_list, sorter_params={}, mode='keep',
-                                        engine='loop', engine_kargs={}):
+                    engine='loop', engine_kargs={}, debug=False):
         run_study_sorters(self.study_folder, sorter_list, sorter_params=sorter_params,
-                                    engine=engine, engine_kargs=engine_kargs)
+                          engine=engine, engine_kargs=engine_kargs, debug=debug)
 
-    def get_ground_truth(self, rec_name):
-        sorting = se.NpzSortingExtractor(self.study_folder / 'ground_truth' / (rec_name+'.npz'))
+    def get_ground_truth(self, rec_name=None):
+        self.scan_folder()
```

Comment:
I don't think we must scan the folder each time.
This can be slow for very big study.

Reply:
Ok, but can one chec if the folder has been scanne dor not? I can make a flag :)

Also, why does the static method `setup` returns: `return cls(study_folder)`. I think it could return nothing.

Reply:
This is class method to setup:
 
So the first ttime to make the folrder:
  * study = GroundTruthStudy.setup(...)

And then to read/compute/compare:
  * study = GroundTruthStudy(folder)